### PR TITLE
Added my function, 2 unittest for my first solution on weight conversion

### PR DESCRIPTION
**Pull request by olumide-AI**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

**Comment by olumide-AI** [comment]
I have two unit test which is unsual. but Evan Cole approved it. Here's evan's message as you review my code. Thank you 
This is great!  You did something different from the challenges I expected, but you did it well.  So you can ignore the file name requirement in code review
you grouped two related (well-named) functions into a clearly named module, which is how software should be done.  Placing the conversion constants in globals is also a nice touch
But I would recommend you change your tests.  You can have one test class for each function (TestPoundsToKilograms and TestKilogramsToPounds).  then each function can have it’s own unit tests organized by edge cases/assertions/… like you did (great job again).
Refactoring your unit tests like this will also help you have only 1 assertion per method (edited)

**Review by Mithchell509 — CHANGES_REQUESTED** [review]
test_negative_integer and test_negative_float should not be in defensives tests ?

File: solutions/tests/test_kilograms_to_pounds.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    def test_negative_float_number(self):
+        """Test conversion of a negative float number """
+        self.assertEqual(kilograms_to_pounds(-2.205), -4.861)
+
+    def test_extremely_large_number(self):
+        """Test conversion of an extremely large number"""
+        self.assertEqual(kilograms_to_pounds(100000000), 220462000.0)
+
+    def test_extremely_small_number(self):
+        """TTest conversion of an extremely small number"""
```
Comment (Alemayehu-Desta) [inline solutions/tests/test_kilograms_to_pounds.py:59]: Typo error, "TTest instead of "Test"

File: solutions/tests/test_kilograms_to_pounds.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        self.assertEqual(kilograms_to_pounds(0.000000001), 0.0)
```
Comment (Alemayehu-Desta) [inline solutions/tests/test_kilograms_to_pounds.py:60]: The test self.assertEqual() to extremly small numbers will round to 0.0, which might not align with the function’s behavior. The code rounds to 3 decimal numbers 

I have learned that you can use assertAlmostEqual() for precision.

File: solutions/tests/test_pounds_to_kilogram.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        """Test conversion of an extremely large number"""
+        self.assertEqual(pounds_to_kilograms(220462000.0), 99999799.504)
+
+    def test_extremely_small_number(self):
+        """Test extremely small number"""
+        self.assertEqual(pounds_to_kilograms(0.00002205), 0.0)
+
+    def test_exponential_number(self):
+        """Test conversion of exponential numbers"""
+        self.assertEqual(pounds_to_kilograms(1*2^-500), -225.889)
```
Comment (Alemayehu-Desta) [inline solutions/tests/test_pounds_to_kilogram.py:64]: The test self.assertEqual(pounds_to_kilograms(1*2^-500), -225.889) is incorrect due to invalid Python syntax (2^-500).

If the code is written in Python, the syntax in Python is self.assertEqual(pounds_to_kilograms(1 * (2 ** -500)), 0.0). Research more on that

**Review by Alemayehu-Desta — DISMISSED** [review]
Hello Olumide,

You did a great job. 

The code and its documentation is comprehensive and clear. Its also well structured to understand the code.

I have left minor comments that I came across in reviewing the code. I recommend you to research on the AssertAlmostEqual() function to handle the extreme cases in your case. 

Great work once again

**Review by Mithchell509 — APPROVED** [review]

File: solutions/tests/test_kilograms_to_pounds.py (hunk 1 above)
Comment (olumide-AI) [reply solutions/tests/test_kilograms_to_pounds.py:59]: Thank you i have resolved that. Good catch

File: solutions/tests/test_kilograms_to_pounds.py (hunk 2 above)
Comment (olumide-AI) [reply solutions/tests/test_kilograms_to_pounds.py:60]: Nice, i will refactor that

**Review by Mithchell509 — APPROVED** [review]

**Review by Alemayehu-Desta — APPROVED** [review]
This implementation fulfills all functional, stylistic, and documentation requirements.
- Clear docstrings and comments.
- Extensive test cases for various scenarios.
- Error handling for edge cases.
- Adherence to coding standards
